File: Field.cpp
#include "Field.hpp"

CoordinateTableSudoku Field::translatorIndexInRowCol(int indexSudokuMassive) {
	CoordinateTableSudoku coordinateCell;
	coordinateCell.setPosition_x(indexSudokuMassive % SIZE_SUDOKU);
	coordinateCell.setPosition_y(indexSudokuMassive / SIZE_SUDOKU);
	return coordinateCell;
}

void Field::InitializeQuantityEmptyValues() {
	memset(quantityValues, 0, sizeof(quantityValues));
}

void Field::InitializeFieldEmptyValues() {
	for (int i = 0; i < SIZE_SUDOKU_N_X_N; i++) {
		field[i].setValue(0);
		field[i].setIs_fixed(true);
	}
}

// Временно
// ---------------
Field::Field(const Field& other) {
	*this = other;
}
// ---------------

File: Sudoku_project/app/src/main/java/com/example/sudoku/core/Field.java
package com.example.sudoku.core;

public class Field {
    public Cell[] field = new Cell[Config.SIZE_SUDOKU_N_X_N];
    public int[] quantityValues = new int[Config.SIZE_COUNTER_IMPUT_NUMS];

    public Field() {
        initializeFieldEmptyValues();
        initializeQuantityEmptyValues();
    }

    public void initializeFieldEmptyValues() {
        for (int i = 0; i < field.length; i++) {
            field[i] = new Cell();
        }
    }

    public void initializeQuantityEmptyValues() {
        for (int i = 0; i < quantityValues.length; i++) {
            quantityValues[i] = 0;
        }
    }

    public CoordinateTableSudoku translatorIndexInRowCol(int index) {
        int y = index / Config.SIZE_SUDOKU;
        int x = index % Config.SIZE_SUDOKU;
        return new CoordinateTableSudoku(x, y);
    }
}
